lib/templates/lab2/lab2.dart

```diff
@@ -1,15 +1,95 @@
-class Book {
-  // your code here
+class Book with Readable {
+  String title, author;
+  int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+
+  void bookDetails() {
+    print('Book Details:\n'
+        'Title: $title\n'
+        'Author: $author\n'
+        'Number of Pages: $numberOfPages\n');
+  }
+}
+
+class FictionBook extends Book with Readable {
```

Comment:
why we need to duplicate readable if it is already in base book class?

Reply:
Because I thought mixin works only with class in which we add it
Is this such a big mistake that 20% of grade is deducted for it?

Reply:
We have 5 tasks, 1 incorrect, so 4/5. It is just 1% of course